### PR TITLE
Test: Added E2E "Modify field within manifest".

### PR DESCRIPTION
**Pull request by Ealianis**
### Description of your changes
- Added new E2E test that takes an existing applied resource, locates its owner AppliedWork, pulls the Work resource from the Hub, locate the applied resource's manifest by ordinal, updates it, and expects the spoke to reapply the update manifest.

I have:
- [x] Read and followed Caravel's [Code of conduct](https://github.com/Azure/k8s-work-api/blob/master/code-of-conduct.md).
- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested
Local e2e test execution.

### CODE REVIEW

**Review by aagusuab — APPROVED** [review]
LGTM

File: tests/e2e/apply_test.go [hunk 1]
```diff
@@ -167,19 +178,72 @@ var _ = ginkgo.Describe("Apply Work", func() {
 				return garbageCollectionComplete
 			}, eventuallyTimeout, eventuallyInterval).Should(gomega.BeTrue())
 		})
+	})
+	ginkgo.Context("Work created on the Hub, then a manifest is modified on the Hub.", func() {
+		ginkgo.It("Should reapply the manifest.", func() {
```
Comment (ryanzhang-oss) [inline tests/e2e/apply_test.go:183]: why?

Comment (ryanzhang-oss) [reply tests/e2e/apply_test.go:183]: also, did we say that we . import ginkgo/gomega?

Comment (Ealianis) [reply tests/e2e/apply_test.go:183]: Discussed with you in Teams chat, the setup of the tests are within the It blocks due to the clients being nil when the test runner is inside the context blocks. It was something I was looking to resolve but this approach was to get the tests defined , and working.

As for the import, are you referring to the OSS form we had to fill out? Ginkgo / gomega are already imported within the upstream source. We didn't add it within our fork.

**Pull request merged**